sdk/resourcemanager/azure-resourcemanager-privatedns/src/test/java/com/azure/resourcemanager/privatedns/PrivateDnsZoneCnameRecordSetTests.java

```diff
@@ -0,0 +1,109 @@
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.management.Region;
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.resourcemanager.privatedns.models.CnameRecordSet;
+import com.azure.resourcemanager.privatedns.models.PrivateDnsZone;
+import com.azure.resourcemanager.resources.ResourceManager;
+import com.azure.resourcemanager.resources.fluentcore.utils.HttpPipelineProvider;
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
+import com.azure.resourcemanager.test.ResourceManagerTestBase;
+import com.azure.resourcemanager.test.utils.TestDelayProvider;
+import com.azure.resourcemanager.test.utils.TestUtilities;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+public class PrivateDnsZoneCnameRecordSetTests extends ResourceManagerTestBase {
+    private String rgName = "";
+
+    protected ResourceManager resourceManager;
+    protected PrivateDnsZoneManager privateZoneManager;
+
+    @Override
+    protected HttpPipeline buildHttpPipeline(
+        TokenCredential credential,
+        AzureProfile profile,
+        HttpLogOptions httpLogOptions,
+        List<HttpPipelinePolicy> policies,
+        HttpClient httpClient) {
+        return HttpPipelineProvider.buildHttpPipeline(
+            credential,
+            profile,
+            null,
+            httpLogOptions,
+            null,
+            new RetryPolicy("Retry-After", ChronoUnit.SECONDS),
+            policies,
+            httpClient);
+    }
+
+
+
+    @Override
+    protected void initializeClients(HttpPipeline httpPipeline, AzureProfile profile) {
+        ResourceManagerUtils.InternalRuntimeContext.setDelayProvider(new TestDelayProvider(!isPlaybackMode()));
+        privateZoneManager = buildManager(PrivateDnsZoneManager.class, httpPipeline, profile);
+        resourceManager = privateZoneManager.resourceManager();
+        rgName = generateRandomResourceName("prdncsrstest", 15);
+    }
+
+    @Override
+    protected void cleanUpResources() {
+        resourceManager.resourceGroups().beginDeleteByName(rgName);
+    }
+
+    @Test
+    public void canUpdateCname() {
+        final Region region = Region.US_EAST;
+        final String topLevelDomain = "www.contoso" + generateRandomResourceName("z", 10) + ".com";
+
+        PrivateDnsZone dnsZone =
+            privateZoneManager
+                .privateZones()
+                .define(topLevelDomain)
+                .withNewResourceGroup(rgName, region)
+                .defineCnameRecordSet("www")
+                    .withAlias("cname.contoso.com")
+                    .withTimeToLive(7200)
+                    .attach()
+                .create();
+
+        // Check CNAME records
+        dnsZone.refresh();
```

Comment:
I think `refresh()` here is a little obsolete since `innerModel()` will be refreshed by calling `create()` or `update()`.

Reply:
Likely dev just calls it to ensure the object is synced with server.

Reply:
Yes, I have just copied other tests as closely as possibly. Let me know if I should remove this (I guess that means the http playback must be updated as well).

Reply:
This should be fine, no need to change.